src/string_util.cc

```diff
@@ -160,13 +160,19 @@ std::string StrFormat(const char* format, ...) {
   return tmp;
 }
 
-void ReplaceAll(std::string* str, const std::string& from,
-                const std::string& to) {
-  std::size_t start = 0;
-  while ((start = str->find(from, start)) != std::string::npos) {
-    str->replace(start, from.length(), to);
-    start += to.length();
-  }
+std::string StrEscape(const std::string & s) {
+  std::string tmp;
```

Comment:
tmp.reserve(s.size());

Reply:
feel free to take over the PR

Reply:
I mean, it's your PR.

Reply:
my use case isn't going to notice the 2 ns (maybe?) speed improvement, if you need it feel free to add it though.